packages/experimental/theme/src/themeReference.ts

```diff
@@ -0,0 +1,101 @@
+import { Theme, PartialTheme } from '@fluentui-react-native/theme-types';
+import { mergeTheme } from './mergeTheme';
+
+/**
+ * Argument type for listening for theme changes
+ */
+export type OnThemeChange = () => void;
+
+/**
+ * A function that takes in a parent theme, then returns a partial set of changes to be merged with the parent theme to update
+ * it.
+ */
+export type ThemeTransform<TTheme extends Theme, TPartial extends PartialTheme> = (parent: TTheme) => TPartial;
+
+/**
+ * A theme recipe is either a partial theme to merge on top of a base theme, or a function that returns the same kind of
+ * partial to merge. These are recipes because they are cached as distinct steps so that they can be reapplied if the
+ * base theme changes
+ */
+export type ThemeRecipe<TTheme extends Theme, TPartial extends PartialTheme> = TPartial | ThemeTransform<TTheme, TPartial>;
+
+/**
+ * Provides a wrapper around a theme object, allowing the ability to create a theme from another theme, to listen to theme
+ * changes, and to update the theme.
+ *
+ * This is the object that is designed to be handed to a theme provider, such that it can listen for changes and set the
+ * value into context again.
+ */
+export class ThemeReference<TTheme extends Theme = Theme, TPartial extends PartialTheme = PartialTheme> {
+  private themeData: TTheme | undefined = undefined;
+  private recipes: ThemeRecipe<TTheme, TPartial>[];
+  private listeners: OnThemeChange[] = [];
+  private getParent: () => TTheme;
+  private parentRef: ThemeReference<TTheme, TPartial> | undefined = undefined;
+
+  /**
+   * Create the theme reference, either as a plain wrapper, or a wrapper with additional transforms and/or merging
+   * @param base - can be either another ThemeReference object, or a fully specified theme
+   * @param recipes - any number of recipies to be applied on top of the theme object
+   */
+  constructor(base: TTheme | ThemeReference<TTheme, TPartial>, ...recipes: ThemeRecipe<TTheme, TPartial>[]) {
```

Comment:
Do we want to export some sort of baselineTheme ThemeReference?

Reply:
Neveeeeermind, I was just trying to avoid the boilerplate of 

`new ThemeReference(getBaselinePlatformTheme(), createOfficeTheme(), myThemeRecipe);`

but then I realized/remembered the createOfficeTheme was a ThemeReference not a ThemeRecipe and  is already layered on the default-theme.

I do wonder if we should take an optional baseThemeReference in createOfficeTheme in case they want to change that though,

Reply:
I'm not sure how common that scenario is in practice though.

Reply:
It's an interesting idea but I'm not sure how common it is in practice as well. It's easy enough to do, and we can do it later without changing the interface by factoring out a helper and exporting the more generic helper.  So I'm going to hold off for now.